ble-shelly-btn.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * This script uses the BLE scan functionality in scripting
+ * Selects Shelly BLU Buttons from the aired advertisements, decodes
+ * the service data payload and toggles a relay on the device on
+ * button push
+ */
+
+// Shelly BLU devices:
+// SBBT - Shelly BLU Button
+// SBDW - Shelly BLU DoorWindow
+
+// sample Shelly DW service_data payload
+// 0x40 0x00 0x4E 0x01 0x64 0x05 0x00 0x00 0x00 0x2D 0x01 0x3F 0x00 0x00
+
+// First byte: BTHome device info, 0x40 - no encryption, BTHome v.2
+// bit 0: “Encryption flag”
+// bit 1-4: “Reserved for future use”
+// bit 5-7: “BTHome Version”
+
+// AD 0: PID, 0x00
+// Value: 0x4E
+
+// AD 1: Battery, 0x01
+// Value, 100%
+
+// AD 2: Illuminance, 0x05
+// Value: 0
+
+// AD 3: Window, 0x2D
+// Value: true, open
+
+// AD 4: Rotation, 0x3F
+// Value: 0
+
+// Device name can be obtained if an active scan is performed
+// You can rely only on the address filtering and forego device name matching
+
+// CHANGE HERE
+function onButtonPress(BTHparsed, BLEData) {
+  print("Button pressed, emitting event");
+  Shelly.emitEvent("BLU_BUTTON", {
+    addr: BLEData.addr,
+    rssi: BLEData.rssi,
+    Button: BTHparsed.Button,
+    Battery: BTHparsed.Battery,
+  });
+}
+
+// remove name prefix to not filter by device name
+// remove address to not filter by address
+let CONFIG = {
+  shelly_blu_name_prefix: "SBBT",
+  //shelly_blu_address: "bc:02:6e:c3:c8:b9",
+  actions: [
+    {
+      cond: {
+        Button: 1,
+      },
+      action: onButtonPress,
+    },
+  ],
+};
```

Comment:
The way this script is structured only allows usage with a single button, but very little extra effort is needed to support multiple buttons with unique actions.